### ISSUE TITLE
Support live form updates

### ISSUE THREAD

**Issue by smpallen99** [ex-admin/ex_admin#16]
As mentioned in #15, I'll be looking to use ExAdmin with my existing live form updates.

**Comment by tmbb** [comment]
How "live" should be form updates? You can have:

1. forms being update when another client submits (easy, just send the new html with populated from the server, and set innerHTML on the JS side- effectively a page reload) or

2. Forma updates keystroke by keystroke, google docs style. Cool but requires instrumenting all text fields with JS for this to work. Cosemirror (plaintext) and quilljs (rich text) already provide enough instrumentation to make it easy to implement realtime collaboration.

Which one of this options are you suggesting? Please not that the naïve solution to option 1 breaks option 2

**Comment by smpallen99** [comment]
When I say live updates, I mean that fields changes will be saved on a field by field basis. Not that the form will be updated if 2 browsers are updating the same form at the same time.

**Comment by tmbb** [comment]
With a save button for every field? Or each time the input changes?

**Comment by smpallen99** [comment]
A live updated form will provide the following:

* When you start editing a text input, a save and cancel button will be shown and CR will save the field.
* radio/checkboxs will save when clicked
* select will save when option is changes
* etc.

**Comment by smpallen99** [comment]
Of course, this will be an optional feature, when defining the form. It will be opt in.

**Comment by tmbb** [comment]
Making it opt in is smart. It avoids breaking changeser validation rules that depend on multiple fields.
